### PR TITLE
Updated code related to bumped deps.

### PR DESCRIPTION
**Pull request by umbreak**

### CODE REVIEW

File: src/main/resources/app.conf [hunk 1]
```diff
@@ -60,22 +60,29 @@ app {
...
-    batch = ${?INDEXING_BATCH}
+    batch-chunk = 10
+    batch-chunk = ${?INDEXING_BATCH}
     # maximum amount of time to wait for the number of events to be taken on each batch
     batch-timeout = 500ms
     batch-timeout = ${?INDEXING_BATCH_TIMEOUT}
     # indexing retry configuration
     retry {
+      # the retry strategy to use; possible values are: "never", "once" and "exponential"
+      strategy = "exponential"
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `INDEXING_RETRY_STRATEGY`

File: src/main/resources/app.conf [hunk 2]
```diff
@@ -60,22 +60,29 @@ app {
...
+      # the initial delay before retrying that will be multiplied with the 'factor' for each attempt
+      # (applicable only for strategy "exponential")
+      initial-delay = 200 milliseconds
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `INDEXING_RETRY_INITIAL_DELAY`

File: src/main/resources/app.conf [hunk 3]
```diff
@@ -60,22 +60,29 @@ app {
...
+      # the maximum delay applied to the retrying strategy
+      max-delay = 10 minutes
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `INDEXING_RETRY_MAX_DELAY`

File: src/main/resources/app.conf [hunk 4]
```diff
@@ -60,22 +60,29 @@ app {
...
+      max-retries = 10
+      max-retries = ${?INDEXING_RETRY_MAX_COUNT}
+
       # the jitter added between retries
-      random-factor = 0.2
-      random-factor = ${?INDEXING_RETRY_RANDOM_FACTOR}
+      factor = 0.2
+      factor = ${?INDEXING_RETRY_RANDOM_FACTOR}
+      # the linear increment (applicable only for strategy "linear")
+      increment = 500 milliseconds
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `INDEXING_RETRY_INCREMENT`

File: src/main/resources/app.conf [hunk 5]
```diff
@@ -113,12 +120,16 @@ app {
       # (applicable only for strategy "exponential")
       initial-delay = 100 milliseconds
       initial-delay = ${?SOURCING_RETRY_INITIAL_DELAY}
+      # the maximum delay applied to the retrying strategy
+      max-delay = 10 minutes
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `SOURCING_RETRY_MAX_DELAY`

File: src/main/resources/app.conf [hunk 6]
```diff
@@ -113,12 +120,16 @@ app {
...
       # maximum number of retries in case of failure (applicable only for strategy "exponential")
       max-retries = 7
       max-retries = ${?SOURCING_RETRY_MAX_RETRY}
       # the exponential factor (applicable only for strategy "exponential")
-      factor = 2
+      factor = 0.2
       factor = ${?SOURCING_RETRY_FACTOR}
+      # the linear increment (applicable only for strategy "linear")
+      increment = 500 milliseconds
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `SOURCING_RETRY_INCREMENT`

File: src/main/resources/app.conf [hunk 7]
```diff
@@ -138,12 +149,16 @@ app {
       # the initial delay before retrying that will be multiplied with the 'factor' for each attempt
       initial-delay = 100 milliseconds
       initial-delay = ${?KEY_VALUE_STORE_RETRY_INITIAL_DELAY}
+      # the maximum delay applied to the retrying strategy
+      max-delay = 10 minutes
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `KEY_VALUE_STORE_RETRY_MAX_DELAY`

File: src/main/resources/app.conf [hunk 8]
```diff
@@ -138,12 +149,16 @@ app {
...
       # maximum number of retries in case of failure (applicable only for strategy "exponential")
       max-retries = 7
       max-retries = ${?KEY_VALUE_STORE_RETRY_MAX_RETRY}
       # the exponential factor (applicable only for strategy "exponential")
-      factor = 2
+      factor = 0.2
       factor = ${?KEY_VALUE_STORE_RETRY_FACTOR}
+      # the linear increment (applicable only for strategy "linear")
+      increment = 500 milliseconds
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `KEY_VALUE_STORE_RETRY_INCREMENT`

File: src/main/resources/app.conf [hunk 9]
```diff
@@ -181,34 +196,45 @@ app {
         # the initial delay before retrying that will be multiplied with the 'factor' for each attempt
         initial-delay = ${app.sourcing.retry.initial-delay}
         initial-delay = ${?ACLS_SOURCING_RETRY_INITIAL_DELAY}
+        # the maximum delay applied to the retrying strategy
+        max-delay = ${app.sourcing.retry.max-delay}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `ACLS_SOURCING_RETRY_MAX_DELAY`

File: src/main/resources/app.conf [hunk 10]
```diff
@@ -181,34 +196,45 @@ app {
...
         # maximum number of retries in case of failure (applicable only for strategy "exponential")
         max-retries = ${app.sourcing.retry.max-retries}
         max-retries = ${?ACLS_SOURCING_RETRY_MAX_RETRY}
         # the exponential factor (applicable only for strategy "exponential")
         factor = ${app.sourcing.retry.factor}
         factor = ${?ACLS_SOURCING_RETRY_FACTOR}
+        # the linear increment (applicable only for strategy "linear")
+        increment = ${app.sourcing.retry.increment}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `ACLS_SOURCING_RETRY_INCREMENT`

File: src/main/resources/app.conf [hunk 11]
```diff
@@ -181,34 +196,45 @@ app {
...
-      batch = ${?ACLS_INDEXING_BATCH}
+      batch-chunk = ${app.indexing.batch-chunk}
+      batch-chunk = ${?ACLS_INDEXING_BATCH}
       # maximum amount of time to wait for the number of events to be taken on each batch
       batch-timeout = ${app.indexing.batch-timeout}
       batch-timeout = ${?ACLS_INDEXING_BATCH_TIMEOUT}
       # indexing retry configuration
       retry {
+        # the retry strategy to use; possible values are: "never", "once" and "exponential"
+        strategy = "exponential"
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `ACLS_INDEXING_RETRY_STRATEGY`

File: src/main/resources/app.conf [hunk 12]
```diff
@@ -181,34 +196,45 @@ app {
...
+        # the initial delay before retrying that will be multiplied with the 'factor' for each attempt
+        # (applicable only for strategy "exponential")
+        initial-delay = ${app.indexing.retry.initial-delay}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `ACLS_INDEXING_RETRY_INITIAL_DELAY`

File: src/main/resources/app.conf [hunk 13]
```diff
@@ -181,34 +196,45 @@ app {
...
+        # the maximum delay applied to the retrying strategy
+        max-delay = ${app.indexing.retry.max-delay}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `ACLS_INDEXING_RETRY_MAX_DELAY`

File: src/main/resources/app.conf [hunk 14]
```diff
@@ -181,34 +196,45 @@ app {
...
+        max-retries = ${app.indexing.retry.max-retries}
+        max-retries = ${?ACL_INDEXING_MAX_RETRIES}
+
         # the jitter added between retries
-        random-factor = ${app.indexing.retry.random-factor}
-        random-factor = ${?ACLS_INDEXING_RETRY_RANDOM_FACTOR}
+        factor = ${app.indexing.retry.factor}
+        factor = ${?ACL_INDEXING_RETRY_RANDOM_FACTOR}
+        # the linear increment (applicable only for strategy "linear")
+        increment = ${app.sourcing.retry.increment}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `ACLS_INDEXING_RETRY_INCREMENT`

File: src/main/resources/app.conf [hunk 15]
```diff
@@ -278,12 +304,16 @@ app {
         # the initial delay before retrying that will be multiplied with the 'factor' for each attempt
         initial-delay = ${app.sourcing.retry.initial-delay}
         initial-delay = ${?PERMISSIONS_SOURCING_RETRY_INITIAL_DELAY}
+        # the maximum delay applied to the retrying strategy
+        max-delay = ${app.sourcing.retry.max-delay}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `PERMISSIONS_SOURCING_RETRY_MAX_DELAY`

File: src/main/resources/app.conf [hunk 16]
```diff
@@ -278,12 +304,16 @@ app {
...
         # maximum number of retries in case of failure (applicable only for strategy "exponential")
         max-retries = ${app.sourcing.retry.max-retries}
         max-retries = ${?PERMISSIONS_SOURCING_RETRY_MAX_RETRY}
         # the exponential factor (applicable only for strategy "exponential")
         factor = ${app.sourcing.retry.factor}
         factor = ${?PERMISSIONS_SOURCING_RETRY_FACTOR}
+        # the linear increment (applicable only for strategy "linear")
+        increment = ${app.sourcing.retry.increment}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `PERMISSIONS_SOURCING_RETRY_INCREMENT`

File: src/main/resources/app.conf [hunk 17]
```diff
@@ -278,12 +304,16 @@ app {
         # the initial delay before retrying that will be multiplied with the 'factor' for each attempt
         initial-delay = ${app.sourcing.retry.initial-delay}
         initial-delay = ${?PERMISSIONS_SOURCING_RETRY_INITIAL_DELAY}
+        # the maximum delay applied to the retrying strategy
+        max-delay = ${app.sourcing.retry.max-delay}
         # maximum number of retries in case of failure (applicable only for strategy "exponential")
         max-retries = ${app.sourcing.retry.max-retries}
         max-retries = ${?PERMISSIONS_SOURCING_RETRY_MAX_RETRY}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: rename env to: `PERMISSIONS_SOURCING_RETRY_MAX_RETRIES`

File: src/main/resources/app.conf [hunk 18]
```diff
@@ -181,34 +196,45 @@ app {
         # the initial delay before retrying that will be multiplied with the 'factor' for each attempt
         initial-delay = ${app.sourcing.retry.initial-delay}
         initial-delay = ${?ACLS_SOURCING_RETRY_INITIAL_DELAY}
+        # the maximum delay applied to the retrying strategy
+        max-delay = ${app.sourcing.retry.max-delay}
         # maximum number of retries in case of failure (applicable only for strategy "exponential")
         max-retries = ${app.sourcing.retry.max-retries}
         max-retries = ${?ACLS_SOURCING_RETRY_MAX_RETRY}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: rename env to: `ACLS_SOURCING_RETRY_MAX_RETRIES`

File: src/main/resources/app.conf [hunk 19]
```diff
@@ -322,12 +352,16 @@ app {
         # the initial delay before retrying that will be multiplied with the 'factor' for each attempt
         initial-delay = ${app.sourcing.retry.initial-delay}
         initial-delay = ${?REALMS_SOURCING_RETRY_INITIAL_DELAY}
+        # the maximum delay applied to the retrying strategy
+        max-delay = ${app.sourcing.retry.max-delay}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: and env override: `REALMS_SOURCING_RETRY_MAX_DELAY`

File: src/main/resources/app.conf [hunk 20]
```diff
@@ -322,12 +352,16 @@ app {
...
         # maximum number of retries in case of failure (applicable only for strategy "exponential")
         max-retries = ${app.sourcing.retry.max-retries}
         max-retries = ${?REALMS_SOURCING_RETRY_MAX_RETRY}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: rename env to: `REALMS_SOURCING_RETRY_MAX_RETRIES`

File: src/main/resources/app.conf [hunk 21]
```diff
@@ -322,12 +352,16 @@ app {
...
         # the exponential factor (applicable only for strategy "exponential")
         factor = ${app.sourcing.retry.factor}
         factor = ${?REALMS_SOURCING_RETRY_FACTOR}
+        # the linear increment (applicable only for strategy "linear")
+        increment = ${app.sourcing.retry.increment}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: add env override: `REALMS_SOURCING_RETRY_INCREMENT`

File: src/main/resources/app.conf [hunk 22]
```diff
@@ -347,34 +381,45 @@ app {
         # the initial delay before retrying that will be multiplied with the 'factor' for each attempt
         initial-delay = ${app.key-value-store.retry.initial-delay}
         initial-delay = ${?REALMS_KEY_VALUE_STORE_RETRY_INITIAL_DELAY}
+        # the maximum delay applied to the retrying strategy
+        max-delay = ${app.sourcing.retry.max-delay}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: update value to: `${app.key-value-store.retry.max-delay}`
and add env override: `REALMS_KEY_VALUE_STORE_RETRY_MAX_DELAY`

File: src/main/resources/app.conf [hunk 23]
```diff
@@ -347,34 +381,45 @@ app {
...
         # maximum number of retries in case of failure (applicable only for strategy "exponential")
         max-retries = ${app.key-value-store.retry.max-retries}
         max-retries = ${?REALMS_KEY_VALUE_STORE_RETRY_MAX_RETRY}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: rename env to: `REALMS_KEY_VALUE_STORE_RETRY_MAX_RETRIES`

File: src/main/resources/app.conf [hunk 24]
```diff
@@ -347,34 +381,45 @@ app {
...
         # the exponential factor (applicable only for strategy "exponential")
         factor = ${app.key-value-store.retry.factor}
         factor = ${?REALMS_KEY_VALUE_STORE_RETRY_FACTOR}
+        # the linear increment (applicable only for strategy "linear")
+        increment = ${app.sourcing.retry.increment}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: update value to `${app.key-value-store.retry.increment}`
and add env override: `REALMS_KEY_VALUE_STORE_RETRY_INCREMENT`

File: src/main/resources/app.conf [hunk 25]
```diff
@@ -347,34 +381,45 @@ app {
...
-      batch = ${?REALMS_INDEXING_BATCH}
+      batch-chunk = ${app.indexing.batch-chunk}
+      batch-chunk = ${?REALMS_INDEXING_BATCH}
       # maximum amount of time to wait for the number of events to be taken on each batch
       batch-timeout = ${app.indexing.batch-timeout}
       batch-timeout = ${?REALMS_INDEXING_BATCH_TIMEOUT}
       # indexing retry configuration
       retry {
+        # the retry strategy to use; possible values are: "never", "once" and "exponential"
+        strategy = "exponential"
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: add env override: `REALMS_INDEXING_RETRY_STRATEGY`

File: src/main/resources/app.conf [hunk 26]
```diff
@@ -347,34 +381,45 @@ app {
...
+        # the initial delay before retrying that will be multiplied with the 'factor' for each attempt
+        # (applicable only for strategy "exponential")
+        initial-delay = ${app.indexing.retry.initial-delay}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: add env override: `REALMS_INDEXING_RETRY_INITIAL_DELAY`

File: src/main/resources/app.conf [hunk 27]
```diff
@@ -347,34 +381,45 @@ app {
...
+        # the maximum delay applied to the retrying strategy
+        max-delay = ${app.indexing.retry.max-delay}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: add env override: `REALMS_INDEXING_RETRY_MAX_DELAY`

File: src/main/resources/app.conf [hunk 28]
```diff
@@ -347,34 +381,45 @@ app {
...
         # the maximum number of times an index function is retried
-        max-count = ${app.indexing.retry.max-count}
-        max-count = ${?REALMS_INDEXING_RETRY_MAX_COUNT}
-        # the maximum amount of time to wait between two retries
-        max-duration = ${app.indexing.retry.max-duration}
-        max-duration = ${?REALMS_INDEXING_RETRY_MAX_DURATION}
+        max-retries = ${app.indexing.retry.max-retries}
+        max-retries = ${?REALMS_ INDEXING_MAX_RETRIES}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: change env to: `REALMS_INDEXING_RETRY_MAX_RETRIES`

File: src/main/resources/app.conf [hunk 29]
```diff
@@ -347,34 +381,45 @@ app {
...
+
         # the jitter added between retries
-        random-factor = ${app.indexing.retry.random-factor}
-        random-factor = ${?REALMS_INDEXING_RETRY_RANDOM_FACTOR}
+        factor = ${app.indexing.retry.factor}
+        factor = ${?REALMS_ INDEXING_RETRY_RANDOM_FACTOR}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: change env to: `REALMS_INDEXING_RETRY_RANDOM_FACTOR`

File: src/main/resources/app.conf [hunk 30]
```diff
@@ -60,22 +60,29 @@ app {
   # default indexing configuration
   indexing {
     # maximum number of events taken on each batch
-    batch = 10
-    batch = ${?INDEXING_BATCH}
+    batch-chunk = 10
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: batch and chunk mean similar things; either keep the name `batch` or rename to `batch-size`

File: src/main/resources/app.conf [hunk 31]
```diff
@@ -181,34 +196,45 @@ app {
...
       }
     }
 
-    # realms indexing configuration
+    # acls indexing configuration
     indexing {
       # maximum number of events taken on each batch
-      batch = ${app.indexing.batch}
-      batch = ${?ACLS_INDEXING_BATCH}
+      batch-chunk = ${app.indexing.batch-chunk}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: batch and chunk mean similar things; either keep the name `batch` or rename to `batch-size`

File: src/main/resources/app.conf [hunk 32]
```diff
@@ -347,34 +381,45 @@ app {
...
+        increment = ${app.sourcing.retry.increment}
       }
     }
 
     # realms indexing configuration
     indexing {
       # maximum number of events taken on each batch
-      batch = ${app.indexing.batch}
-      batch = ${?REALMS_INDEXING_BATCH}
+      batch-chunk = ${app.indexing.batch-chunk}
```
Comment (bogdanromanx) [inline src/main/resources/app.conf]: batch and chunk mean similar things; either keep the name `batch` or rename to `batch-size`

File: src/main/scala/ch/epfl/bluebrain/nexus/iam/realms/Realms.scala [hunk 33]
```diff
@@ -246,7 +248,17 @@ object Realms {
     */
   def index[F[_]: Timer](implicit as: ActorSystem, rc: RealmsConfig, F: Async[F]): RealmIndex[F] = {
     implicit val cfg: KeyValueStoreConfig = rc.keyValueStore
-    KeyValueStore.distributed("realms", (_, resource) => resource.rev, mapError)
+    implicit val monadError: MonadError[F, IamError] = new MonadError[F, IamError] {
```
Comment (bogdanromanx) [inline src/main/scala/ch/epfl/bluebrain/nexus/iam/realms/Realms.scala]: why do we need this additional definition?

Comment (umbreak) [reply src/main/scala/ch/epfl/bluebrain/nexus/iam/realms/Realms.scala]: because we want to retry only on IamError

**Pull request merged**